src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -571,10 +641,56 @@ private unsafe void DetectEvents()
                     bool eventDetected = WasEventDetected(_pollFileDescriptor, -1, out int pinNumber, _eventThreadCancellationTokenSource.Token);
                     if (eventDetected)
                     {
-                        Thread.Sleep(1); // Adding some delay to make sure that the value of the File has been updated so that we will get the right event type.
-                        PinEventTypes eventTypes = (Read(pinNumber) == PinValue.High) ? PinEventTypes.Rising : PinEventTypes.Falling;
-                        var args = new PinValueChangedEventArgs(eventTypes, pinNumber);
-                        _devicePins[pinNumber]?.OnPinValueChanged(args);
+                        if (_statusUpdateSleepTime > 0)
+                        {
+                            Thread.Sleep(_statusUpdateSleepTime); // Adding some delay to make sure that the value of the File has been updated so that we will get the right event type.
+                        }
+
+                        var currentPin = _devicePins[pinNumber];
+                        var activeEdges = currentPin.ActiveEdges;
+                        PinEventTypes eventType = activeEdges;
+                        PinEventTypes secondEvent = PinEventTypes.None;
```

Comment:
Do we really get the secondEvent used a lot? If so, I would question whether we should even handle this case instead of just dropping a few events by design, since I guess in this case you won't really get the second event if there where 3 events between event and read operation, or we won't detect 4 etc.

Reply:
The second event here is guesswork, yes. But it's required to be consistent in the end. 

The logic is as follows: If the previous state is low, we get an event and the current state is still low, we know that there must have been a rise and a fall (otherwise there wouldn't have been an event in the first place). We can't tell whether there have been 4 or 6, but we know we saw an even number of events, which is pretty important for instance if this is connected to a button for instance, where you would only get the raise but not the fall if the contact is short.

Put differently, this should ensure that you don't get two consecutive raises or falls.

Reply:
Oh, and yes: It makes sure that the last reported change corresponds to the current state (that means that the last change was fall when the line stays low for a prolonged time)

Reply:
I see, yeah I see your point and I think it makes sense so it's fine to keep it as is